code/go/0chain.net/blobbercore/allocation/entity.go

```diff
@@ -31,27 +31,28 @@ const (
 )
 
 type Allocation struct {
-	ID             string           `gorm:"column:id;primary_key"`
-	Tx             string           `gorm:"column:tx"`
-	TotalSize      int64            `gorm:"column:size"`
-	UsedSize       int64            `gorm:"column:used_size"`
-	OwnerID        string           `gorm:"column:owner_id"`
-	OwnerPublicKey string           `gorm:"column:owner_public_key"`
-	RepairerID     string           `gorm:"column:repairer_id"` // experimental / blobber node id
-	PayerID        string           `gorm:"column:payer_id"`    // optional / client paying for all r/w ops
-	Expiration     common.Timestamp `gorm:"column:expiration_date"`
+	ID             string           `gorm:"column:id;size:64;primaryKey"`
+	Tx             string           `gorm:"column:tx;size:64;not null;unique;index:idx_unique_allocations_tx,unique"`
+	TotalSize      int64            `gorm:"column:size;not null;default:0"`
+	UsedSize       int64            `gorm:"column:used_size;not null;default:0"`
+	OwnerID        string           `gorm:"column:owner_id;size:64;not null"`
+	OwnerPublicKey string           `gorm:"column:owner_public_key;size:512;not null"`
+	RepairerID     string           `gorm:"column:repairer_id;size:64;not null"`
+	PayerID        string           `gorm:"column:payer_id;size:64;not null"`
+	Expiration     common.Timestamp `gorm:"column:expiration_date;not null"`
 	// AllocationRoot allcation_root of last write_marker
-	AllocationRoot   string        `gorm:"column:allocation_root"`
-	BlobberSize      int64         `gorm:"column:blobber_size"`
-	BlobberSizeUsed  int64         `gorm:"column:blobber_size_used"`
-	LatestRedeemedWM string        `gorm:"column:latest_redeemed_write_marker"`
+	AllocationRoot   string        `gorm:"column:allocation_root;size:64;not null;default:''"`
+	BlobberSize      int64         `gorm:"column:blobber_size;not null;default:0"`
+	BlobberSizeUsed  int64         `gorm:"column:blobber_size_used;not null;default:0"`
+	LatestRedeemedWM string        `gorm:"column:latest_redeemed_write_marker;size:64"`
 	IsRedeemRequired bool          `gorm:"column:is_redeem_required"`
-	TimeUnit         time.Duration `gorm:"column:time_unit"`
-	IsImmutable      bool          `gorm:"is_immutable"`
+	TimeUnit         time.Duration `gorm:"column:time_unit;not null;default:172800000000000"`
```

Comment:
What's the default value 172800000000000 for

Reply:
This is weird, yes. I exactly tried to replicate every thing from .sql files. Hence this code.
We could make further changes in next PR after this one passes as it will be easier to make database-changes then.

@peterlimg  Should I try to make reasonable changes?

Reply:
okay, leave it then, change if necessary in another PR